generate.go

```diff
@@ -70,24 +78,71 @@ func (c *ConfigGenerator) GenerateAnswers() (Answers, error) {
 		// 2. set service links
 		// note that service link overrides the container link (if the names collide)
 		for key, linkAlias := range clientIpsToServiceLinks[clientIp] {
-			if strings.EqualFold(key, linkAlias) {
-				// skip non-aliased service links
-				// they are present in defaults
-				continue
-			}
-			linkedService := svcNameToSvc[key]
-			linkServiceFqdn := getServiceFqdn(&linkedService)
-			globalAliasName := getLinkGlobalFqdn(linkAlias, &linkedService)
-			stackAliasName := getLinkStackFqdn(linkAlias, &linkedService)
-			if _, ok := aRecs[linkServiceFqdn]; ok {
-				//we store 2 A records for link:
-				// a) linkName.namespace
-				// b) linkName.stackName.namespace
-				cARecs[stackAliasName] = aRecs[linkServiceFqdn]
-				cARecs[globalAliasName] = aRecs[linkServiceFqdn]
-			} else if _, ok := cRecs[linkServiceFqdn]; ok {
-				cCnameRecs[stackAliasName] = cRecs[linkServiceFqdn]
-				cCnameRecs[globalAliasName] = cRecs[linkServiceFqdn]
+			splitSvcName := strings.Split(key, "/")
+			if len(splitSvcName) > 2 {
+				linkedService, err := c.metaFetcher.GetService(key)
```

Comment:
I think it might make sense to save services(s) we catch by GetService(key) along with their container, in hashMap map[key]map[service][]containers. In case the same service is linked to multiple client ips, we can save on metadata calls.